lib/datadog/tracing/distributed/fetcher.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+# typed: true
+
+require_relative 'helpers'
+
+module Datadog
+  module Tracing
+    module Distributed
+      # Common fetcher that retrieves fields from a Hash data input
+      class Fetcher
```

Comment:
It could! Any suggestions? 😄
It doesn't know how to set data btw, only read. Setting data is always done into tracer-internal classes (`TraceDigest.new(your,data,here)`), thus doesn't need a custom "Settter" per protocol.

Also, this is completely internal. I'm more than happy to change this in the future.